ls1intum/Artemis
# `Assessment`: Add consistent handling if no more unassessed submissions are available

### Checklist
#### General


#### Server

### Motivation and Context



### Description
The endpoints for getting submission without assessments
- `/exercises/{exerciseId}/modeling-submission-without-assessment`
- `/exercises/{exerciseId}/programming-submission-without-assessment`
- `/exercises/{exerciseId}/text-submission-without-assessment`
- `/exercises/{exerciseId}/file-upload-submission-without-assessment`

used to throw an exception that returns an 404 error to the client.
Since not having anymore unassessed submissions is normal behavior, returning null is a better solution. 

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student

1. instructor: Setup exercises for each type i.e. Programming Exercise, Modeling Exercise, etc.
2. student: Create submissions for each exercise
3. instructor: Check if you can assess/access the submissions for the exercise dashboards.

(Additionally check if you get an error after reaching the last submission)

### Review Progress






#### Code Review
#### Manual Tests




### Screenshots
